### PR TITLE
Pull to refresh

### PR DESCRIPTION
**Pull request by helenxuyang**
### Summary 

This pull request is the first step towards implementing pulling to refresh instead of sending requests every time the nav bar is used

- [x] removed requests when nav bar is used
- [x] added pull to refresh for homepage and history page
- [x] fixed provider and history page so that completed rides will show up on the history page without needing a request

### Test Plan 

Added a new active ride in backend and refreshed: 
https://user-images.githubusercontent.com/60579411/112573278-5853f080-8dc2-11eb-9b82-200c6989744d.mp4

Added a new past ride in backend and refreshed:
https://user-images.githubusercontent.com/60579411/112573794-5cccd900-8dc3-11eb-9468-b2dff8b116b5.mp4

Checked that a ride is added to the history page correctly:
https://user-images.githubusercontent.com/60579411/112574680-327c1b00-8dc5-11eb-97e3-79dc9547be1a.mp4

### CODE REVIEW

File: lib/pages/RideHistory.dart [hunk 1]
```diff
@@ -7,54 +10,75 @@ import 'package:flutter/rendering.dart';
...
-    List<int> days = rideGroups.keys.toList()..sort((a, b) => a - b);
 
     return SafeArea(
-      child: SingleChildScrollView(
-        child: Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
+      child: RefreshIndicator(
+        onRefresh: () async {
+          await ridesProvider.requestPastRides(appConfig, authProvider);
+          setState(() {});
```
Comment (tyang98) [inline lib/pages/RideHistory.dart]: Is setState() necessary if it has nothing in its body?

File: lib/pages/Rides.dart [hunk 2]
```diff
@@ -230,24 +211,59 @@ class _RidesState extends State<Rides> {
...
-      onDropoff: () {
-        setState(() {
-          selectedRides.forEach((Ride r) => finishRide(context, r));
-          selectedRides = [];
-        });
+    return !ridesProvider.hasActiveRides() ? Center(child: CircularProgressIndicator()) :
+    RefreshIndicator(
+      onRefresh: () async {
+        await ridesProvider.requestActiveRides(appConfig, authProvider);
+        setState(() {});
```
Comment (tyang98) [inline lib/pages/Rides.dart]: Same question here as before.

**Review by tyang98 — APPROVED** [review]
In addition to the rides you added for me, I added some more rides and the pages were able to refresh properly. Thus, I think this PR is sound in terms of functionality. In terms of the code, a lot of the diffs seem to be simply changes in the code's formatting.

File: lib/pages/RideHistory.dart (hunk 1 above)
Comment (helenxuyang) [reply lib/pages/RideHistory.dart]: Apparently I still don't understand how providers work--I had thought the setState was needed to force the page to update, but now I'm realizing if we don't put listen: false then it actually does rebuild automatically. Will fix, thanks!

**Pull request merged**